ldms/src/core/ldms.h

```diff
@@ -1381,8 +1381,35 @@ void ldms_qgroup_info_free(ldms_qgroup_info_t qinfo);
 typedef enum ldms_stream_type_e {
 	LDMS_STREAM_STRING,
 	LDMS_STREAM_JSON,
+	LDMS_STREAM_AVRO_SER,
+	LDMS_STREAM_LAST, /* the last enumureation; not a real type */
 } ldms_stream_type_t;
 
+/**
+ * \brief Stream Type Symbol.
+ *
+ * This function returns a constant string symbol (e.g. "LDMS_STREAM_STRING") of
+ * the given Stream type \c t. If the given type is out of valid range,
+ * "UNKNOWN" is returned.
+ *
+ * \param t The Stream type.
+ *
+ * \retval s The symbol (e.g. "LDMS_STREAM_STRING").
+ */
+const char *ldms_stream_type_sym(ldms_stream_type_t t);
+
+typedef struct ldms_stream_client_s *ldms_stream_client_t;
+typedef struct json_entity_s *json_entity_t;
+
+/* from avro/schema.h */
+typedef struct avro_obj_t *avro_schema_t;
```

Comment:
Please don't cut-and-paste definitions from external headers. We should have the proper #include <avro/schema.h> (and other includes) instead.  Maybe that means making the inclusion conditional and/or splitting the public stream API out into a header other than ldms.h.